nucleus/utils.py

```diff
@@ -422,3 +435,120 @@ def paginate_generator(
             yield json_value
         if not next_token:
             break
+
+
+def split_s3_bucket_key(s3_path: str):
+    s3_bucket, s3_key = s3_path.split("//", 1)[-1].split("/", 1)
+    return s3_bucket, s3_key
+
+
+def fetch_image(s3_url: str):
+    s3_bucket, s3_key = split_s3_bucket_key(s3_url)
+    image = Image.open(
+        boto3.resource("s3").Bucket(s3_bucket).Object(s3_key).get()["Body"]
+    )
+    return image
+
+
+def fetch_chip(ref_id: str):
```

Comment:
This needs a docstring, what do all of the permutations of the return values mean? I'd also consider composing the ref_id in this function rather than outside.